planner/core/physical_plans.go

```diff
@@ -545,6 +549,22 @@ type PhysicalWindow struct {
 	Frame           *WindowFrame
 }
 
+// PhysicalPartition represents a partition plan.
+type PhysicalPartition struct {
+	physicalSchemaProducer
+
+	Concurrency int
+	Tail        PhysicalPlan
+	DataSource  PhysicalPlan
+}
+
+// PhysicalPartitionDataSourceStub represents a data source stub of partition plan.
+type PhysicalPartitionDataSourceStub struct {
+	physicalSchemaProducer
```

Comment:
`PhysicalPartitionDataSourceStub` does not change schema for now, so is it better to use `basePhysicalPlan`?

Reply:
`PhysicalPartitionDataSourceStub` has no child, so I think it need storing the schema by itself ?